### ISSUE TITLE
Switch to production

### ISSUE THREAD

**Issue by MiroCillik** [keboola/oauth-api#83]

**Comment by Halama** [comment]
Vypadá to dobře. Pro ten krok 1 je vše ready teda a pouštíš se do něj?

**Comment by MiroCillik** [comment]
Este rozmyslam, ze vyhodim tu podporu pre "custom" oauth credentials https://github.com/keboola/oauth-api/issues/85

**Comment by MiroCillik** [comment]
Este treba pridat AP region do deploy.sh skriptu:
https://github.com/keboola/oauth-api/issues/105

A zaregistrovat callback urls pre vsetky komponenty, to sa musi rucne:
https://github.com/keboola/oauth-api/issues/100

**Comment by Halama** [comment]
@MiroCillik Updatuj tady prosím ten progress. Bavili jsme se o tom ve čtvrtek, částečně je to teda vidět tady https://github.com/keboola/oauth-api/milestone/1
Kromě tech "provozní" issues je teda potřeba připravit to migrování credentials (backend + ui) #84 a podporu v UI https://github.com/keboola/kbc-ui/issues/2065 pro nové UI. 

Postup bude teda?:
- zmigrovat consumers do všech regionů
- při konfigurování oauth v UI používat už nový oauth (možná na první testy chvíli přes feature projektu?)
- upravit v dokumentaci https://sites.google.com/keboola.com/devel-internal/komponenty/oauth/registrace-oauth?pli=1&authuser=1 a úpravy a přidávání nových appek dělat už jenom v novém
- oznámit to na status
- u stávajácích konfigurací nabízet možnost zmigrovat konfigurace na nový oauth pro celou komponentu.
- sledovat používání starého oauth

**Comment by Halama** [comment]
@MiroCillik jak to teď vypadá. Consumers jsou zmigrovaný a vyřešený? To UI už by mohlo být připraveno, jakým způsobem se teda releasne. Přes feature projektu zakládání všech oauth na nové verzi? A mezitím připravovat migraci credentials.

**Comment by MiroCillik** [comment]
@Halama Consumers su zmigrovany v US. Teraz to este nasadzujem do AP a spustim migraci v EU a AP. Ale v tychto dvoch este budem musiet pridat niektorych Consumerov rucne. Tam kde som zakladal nove API creds.
A pak sa hned pustim do UI, spravil by som to na feature projektu zatial. Ja uz to mam na lokaly aj pripravene z casti. Vlastne sa tam len prida ta verzia 3 do konfigu a treba upravit aby sa tie url nacitali z tych services.

**Comment by Halama** [comment]
super, v tom ui je jenom potřeba teda pohlídat aby fungovaly obě ty verze jak přidávání tak to zobrazení a fetch.

**Comment by MiroCillik** [comment]
JJ, a potom mozem zacat pripravovat tu migraci credentials

**Comment by MiroCillik** [comment]
Zalozena project feature pre podmineku v UI:
```json
{
    "id": 90,
    "name": "oauth-v3",
    "type": "project",
    "description": "Projects with this feature will use new OAuth API (v3) for new configurations",
    "created": "2018-11-05T14:06:43+0100"
}
```

**Comment by MiroCillik** [comment]
@Halama ano, ako pises, updatol som stacky na produkci a nasadil to. Cakam na review a pustam sa do migracie credentials.

**Comment by MiroCillik** [comment]
Krok 3. nasadeny ale zatial len pre projekty so zapnutou feature `oauth-v3`.

**Comment by Halama** [comment]
Přepnul jsem si feature na nový oauth a vyzkoušel pár extractorů a writerů. Sledoval jsem jestli to fakt volá nové api apod. Vypadá to že to funguje dobře 👍 

Dál by bylo dobrý myslím teda:
- uzavřít že jsou všechny konsumers ok #82
- připravit ten migrátor credentials
- zmigrovat a nastavit feature např. shared configu nebo nějakým našim produkčním projektům  a sledovat
- novým projektům nastavovat `oauth-v3` feature
- monitorovat přístupy na starý oauth - bylo by super vidět co jaký projekt volá . Mít jistotu že v projektech z předchozího kroku nic nepoleze na starý oauth
- vypnout feature a zapnout nový oauth pro nové komponenty všem a nabízet migraci starých oauth + status

**Comment by Halama** [comment]
Přístupy ke starému oauth jdou vidět tady, bohužel se tam neposílá id projektu ale https://rpm.newrelic.com/accounts/218779/applications/6533656/transactions#id=5b225765625472616e73616374696f6e2f416374696f6e2f6b65626f6f6c615f6f5f617574685f6765745f63726564656e7469616c73222c22225d

**Comment by MiroCillik** [comment]
Consumerov riesim v tomto issue https://github.com/keboola/oauth-api/issues/100. Skusim prejst ten zoznam vo vsetkych regionoch :)

**Comment by Halama** [comment]
Ta feature novým projektům se dá udělat takhle https://keboolamanagementapi.docs.apiary.io/#reference/super-features/project-template-features/add-a-feature Asi se může nastavit všem public templates.

**Comment by MiroCillik** [comment]
Tu je PR pre config-migration-tool pre migraciu Credentials https://github.com/keboola/config-migration-tool/pull/28

**Comment by MiroCillik** [comment]
Ked sa nasadi tento bugfix https://github.com/keboola/kbc-ui/pull/2525, tak zapnem `oauth-v3` featuru aj nove projekty v US.
A budem pracovat na UI pre tu migraciu Credentials.

**Comment by Halama** [comment]
je potřeba před dalšími kroky switchnout na ten nový logger at tam je správně system a program v PT https://github.com/keboola/oauth-api/issues/125
Teď tu vidím např.
![image](https://user-images.githubusercontent.com/903531/50523625-81955180-0ad1-11e9-9432-4c5a15ff8109.png)

což je špatně protože tam není kbc stack.

**Comment by Halama** [comment]
Co znamená tohle? Chybí tam access log a jaká funkce byla spuštěna.
![image](https://user-images.githubusercontent.com/903531/50524008-9e328900-0ad3-11e9-996c-4e1808164a97.png)

**Comment by Halama** [comment]
Uff, už to asi konečně chápu. Tohle jsou logy toho serverless pt loggeru které tam posílá ten globální handler. To co timeoutuje je tedy ta serverless logger lambda https://keboola.slack.com/archives/C02CGK5CL/p1546021698057900 . Takže je potřeba odstranit ten serverless PT logger.

![image](https://user-images.githubusercontent.com/903531/50524957-f61fbe80-0ad8-11e9-8d34-c0453bb83d02.png)

**Comment by MiroCillik** [comment]
OK, idem na to

**Comment by MiroCillik** [comment]
Issue pre kbc-ui https://github.com/keboola/kbc-ui/pull/2624

**Comment by MiroCillik** [comment]
Druha UI issue https://github.com/keboola/kbc-ui/pull/2742

Zatial sa deprecation warning na Overview zobrazi len pre projeky s feature `oauth-migration`.
